sycl/include/CL/sycl/queue.hpp

```diff
@@ -320,17 +320,52 @@ class __SYCL_EXPORT queue {
 
   /// Fills the specified memory with the specified pattern.
   ///
-  /// \param Ptr is the pointer to the memory to fill
+  /// \param Ptr is the pointer to the memory to fill.
   /// \param Pattern is the pattern to fill into the memory.  T should be
   /// trivially copyable.
   /// \param Count is the number of times to fill Pattern into Ptr.
+  /// \return an event representing fill operation.
   template <typename T> event fill(void *Ptr, const T &Pattern, size_t Count) {
     return submit([&](handler &CGH) { CGH.fill<T>(Ptr, Pattern, Count); });
   }
 
+  /// Fills the specified memory with the specified pattern.
+  ///
+  /// \param Ptr is the pointer to the memory to fill.
+  /// \param Pattern is the pattern to fill into the memory.  T should be
+  /// trivially copyable.
+  /// \param Count is the number of times to fill Pattern into Ptr.
+  /// \param DepEvent is an event that specifies the kernel dependencies.
+  /// \return an event representing fill operation.
+  template <typename T>
+  event fill(void *Ptr, const T &Pattern, size_t Count, event DepEvent) {
+    return submit([&](handler &CGH) {
+      CGH.depends_on(DepEvent);
+      CGH.fill<T>(Ptr, Pattern, Count);
+    });
+  }
+
+  /// Fills the specified memory with the specified pattern.
+  ///
+  /// \param Ptr is the pointer to the memory to fill.
+  /// \param Pattern is the pattern to fill into the memory.  T should be
+  /// trivially copyable.
```

Comment:
If the Pattern is supposed to be trivially copyable, should this template be qualified with a `std::enable-if` clause?  

In Section 4.9.4.3 of the SYCL2020 spec it says that "T must be a scalar value or a SYCL vector type." for one of the other variants of `.fill()`     But for this variant (in 4.6.5.2) the spec has a one word link: "USM". 

@Pennycook do you know?

Reply:
The specification doesn't say "Only available if", which is SYCL 2020 speak for "Guarded by enable if".  So by the current specification, no.  @jbrodman, do you think there should be compile-time checking of `T` here?

Reply:
BTW, there is a `static_assert(is_trivially_copyable<T>)` in [the implementation of `handler::fill`](https://github.com/intel/llvm/blob/sycl/sycl/include/CL/sycl/handler.hpp#L2151-L2152), which is called from here.

Reply:
I think the static assert is fine.  I agree with John that enable_if isn't the right thing here.